### PR TITLE
[FINERACT-1678] -  COB Partitioning tasks creates fully balanced id ranges 

### PR DESCRIPTION
**Pull request by reluxa**
- COB Partitioning tasks creates fully balanced id ranges where the number of the loans between the min max are matching with the defined pageSize.

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by adamsaghy** [comment]
@reluxa Please dont forget to squash the commits! ;)

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import org.junit.Ignore;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SuppressWarnings("rawtypes")
+@Slf4j
+@ExtendWith(LoanTestLifecycleExtension.class)
+@Ignore
```
Comment (adamsaghy) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java]: Ignore?

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                    }
+                    countDownLatch.countDown();
+                });
+            }
+            countDownLatch.await();
+            // select loan_status_id, count(*) from m_loan group by loan_status_id;
+
+            log.info("**** Changing business day...");
+            updateBusinessDateAndExecuteCOBJob("2020-03-03");
+            executorService.shutdown();
```
Comment (adamsaghy) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java]: I dont really see any assertion whether the COB was executed properly and the loans were processed... :/

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please kindly see my comments!

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java (hunk 2 above)
Comment (reluxa) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java]: fixed as we agreed

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java (hunk 1 above)
Comment (reluxa) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java]: fixed as we agreed

Comment (adamsaghy) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java]: you might wanna remove the ignore annotation, no? :)

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**